### ISSUE TITLE
Allow for extensibility.

### ISSUE THREAD

**Issue by mplodowski** [rainlab/pages-plugin#113]
In plugin I currently develop I need to extend static page.

**Comment by daftspunk** [comment]
Please provide more detail...

**Comment by mplodowski** [comment]
I need some way to hook into after static page save event. The only way I can think of is to add after save event. I know static page is build on top off CMS page but it does not fire the same events.

Example use for this event:

1. Save custom fields to database. Eg. seo fields, tags, etc. Using viewBag for everything is bad and not allow for polymorphic relations.
2. Integration for third party tools. You can for example have plugin that logs all users actions, that event will allow for that.
3. Extensibility for other developers, who wants to do some modifications how this plugin works.

**Comment by mplodowski** [comment]
@daftspunk do you plan to merge my PR for this?

Do you want more details or informations?

**Comment by FelixINX** [comment]
Why do you need this PR @mplodowski ?

And yes, please provide the more details possible.

**Comment by mplodowski** [comment]
@FelixINX does my previous explanation was not enough?

Here is another try:

In my plugin I added more SEO fields and OpenGraph fields to RainLab.Pages plugin. Those fields are polymorphic and can be used in many other places, so using viewBag is not the best solution. I tried to hook to CMS page after save event, because Static Page derives from CMS Page, but this event is not fired after Static Page is saved.

My [PR](https://github.com/mplodowski/pages-plugin/commit/7831635200d3ad59ece73cbb3e96f3038c78cecf) solves my problem and can be used for examples I mentioned in earlier exaplanation.

I don not know, what more details I can provide. Allow for extension should be the goal for all widely used plugins such as this one.
